itest/rust/src/builtin_tests/script/script_instance_tests.rs

```diff
@@ -0,0 +1,181 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use std::ffi::c_void;
+use std::rc::Rc;
+
+use godot::bind::{godot_api, GodotClass};
+use godot::builtin::meta::{ClassName, MethodInfo, PropertyInfo};
+use godot::builtin::{GString, ScriptInstance};
+use godot::engine::global::{MethodFlags, PropertyHint, PropertyUsageFlags};
+use godot::engine::{IScriptExtension, Script, ScriptExtension, ScriptLanguage};
+use godot::prelude::*;
+use godot::sys;
+
+#[derive(GodotClass)]
+#[class(base = ScriptExtension, init)]
+struct TestScript {
+    base: Gd<ScriptExtension>,
+}
+
+#[godot_api]
+impl IScriptExtension for TestScript {
+    unsafe fn instance_create(&self, _for_object: Gd<Object>) -> *mut c_void {
+        create_script_instance(TestScriptInstance::new(self.base.clone().upcast())) as *mut c_void
+    }
+
+    fn can_instantiate(&self) -> bool {
+        true
+    }
+}
+
+struct TestScriptInstance {
+    prop_b: bool,
+    prop_list: Rc<Vec<PropertyInfo>>,
+    method_list: Rc<Vec<MethodInfo>>,
+    script: Gd<Script>,
+}
+
+impl TestScriptInstance {
+    fn new(script: Gd<Script>) -> Self {
+        Self {
+            script,
+            prop_b: false,
+            prop_list: Rc::new(vec![PropertyInfo {
+                variant_type: VariantType::Int,
+                property_name: StringName::from("script_property_a"),
+                class_name: ClassName::from_ascii_cstr("\0".as_bytes()),
+                hint: PropertyHint::PROPERTY_HINT_NONE,
+                hint_string: GString::new(),
+                usage: PropertyUsageFlags::PROPERTY_USAGE_NONE,
+            }]),
+
+            method_list: Rc::new(vec![MethodInfo {
+                id: 1,
+                method_name: StringName::from("script_method_a"),
+                class_name: ClassName::from_ascii_cstr("TestScript\0".as_bytes()),
+                return_type: PropertyInfo {
+                    variant_type: VariantType::String,
+                    class_name: ClassName::from_ascii_cstr("\0".as_bytes()),
+                    property_name: StringName::from(""),
```

Comment:
Is this a valid property name? Otherwise come up with something random...

also below

Reply:
The engine uses empty string names for a return value `PropertyInfo`. I didn't find a better way to construct an empty `StringName`.